src/AntiCorruptionLayer/AntiCorruptionMessageDispatcher.php

```diff
@@ -24,9 +25,11 @@ public function __construct(
         $this->filterAfter = $filterAfter ?? new AllowAllMessages();
     }
 
-    public function dispatch(Message ...$messages): void
+    public function dispatch(iterable|Message $messages): void
     {
         $forwarded = [];
```

Comment:
To continue to leverage generators, maybe $forwarded could always be a generator?

```suggestion
        
    $messages = is_iterable($messages) ? $messages : [$messages];

    $forwarded = (function () use ($messages): Generator {
        foreach ($messages as $message) {
            if (! $this->filterBefore->allows($message)) {
                continue;
            }
            
            $message = $this->translator->translateMessage($message);
            
            if (! $this->filterAfter->allows($message)) {
                continue;
            }
            
            yield $message;
        }
    })();
```